src/compas/geometry/pointclouds/pointcloud.py

```diff
@@ -158,6 +158,25 @@ def __setitem__(self, key, value):
     def __iter__(self):
         return iter(self.points)
 
+    def __eq__(self, other):
+        """Is this pointcloud equal to the other pointcloud?
+
+        Two pointclouds are considered equal if they have the same number of points
+        and if the XYZ coordinates of the corresponding points are identical.
+
+        Parameters
+        ----------
+        other : :class:`compas.geometry.Pointcloud` or list
+            The pointcloud to compare.
+
+        Returns
+        -------
+        bool
+            True if the pointclouds are equal.
+            False otherwise.
+        """
+        return all(a == b for a, b in zip(self, other))
```

Comment:
This doesn't check that the pointclouds have the same number of points.

Reply:
true. will update

Reply:
Also, does the order of the points in a pointcloud really matter?

Reply:
no it doesn't, but couldn't think of a different comparison. we could compare max distance between nearest neighbours?

Reply:
Yeah, I think lexicographically sorting the points and then comparing them pairwise would make sense here.

Reply:
could do something like this
```python
import random
from compas.geometry import Pointcloud
a = Pointcloud.from_bounds(10, 10, 10, 10)
points = a.points[:]
random.shuffle(points)
b = Pointcloud(points)
a_points = sorted(a.points, key=lambda point: (point.x, point.y, point.z))
b_points = sorted(b.points, key=lambda point: (point.x, point.y, point.z))
```
and then compare?

Reply:
yeah, that looks good.